### PR TITLE
PG VS normalise addons test

### PR DESCRIPTION
**Pull request by danoli3**
Testing actions for normalise path

### CODE REVIEW

**Comment by danoli3** [comment]
Having persistent issue with MSVC with error C2665. Essentially filesystem or string can't be used in same function overload. 
There is no link to the direct cause. I've manually been changing each function overload to try and overcome this or determine the issue...

@dimitre any ideas, feel free to check this branch out and have a look. No issues on macOS or linux though

**Comment by danoli3** [comment]
Okay I think I fixed it. macOS then was having issues with ctor / copy ctor. 

Made a bit of mess of ofAddons in fixing this with overloaded functions to unique ones... 

fs::path / string are fine as function prototypes... but since fs::path directly addressed, resolves to to the string, functions don't know which one to use in MSVC so became an issue.

**Comment by danoli3** [comment]
macOS working 
Testing VS

**Comment by danoli3** [comment]
```
Command failed: "C:\srx\openFrameworks\projectGenerator\resources\app\app\projectGenerator"  -o"C:\srx\openFrameworks\" -a" " -p"vs,vs"  -t"" "C:\srx\openFrameworks\examples\templates\emptyExample\allAddonsExample"
[ error ] { "errorMessage": "fs::create_directory failed, "projectPath": { "C:\srx\openFrameworks\projectGenerator\ -pvs,vs  -t" C:\srx\openFrameworks\examples\templates\emptyExample\allAddonsExample" }, "exception": "create_directories: The filename, directory name, or volume label syntax is incorrect.: "C:\srx\openFrameworks\projectGenerator\ -pvs,vs  -t" C:\srx\openFrameworks\examples\templates\emptyExample\allAddonsExample""", "status:" "EXIT_FAILURE" }
```

**Comment by danoli3** [comment]
Traced to optionsparser.h 
Patched now in src here : 
Version 1.7: Work on const-correctness.
**Version 1.6: Fix for MSC compiler.**
Version 1.5: Fixed 2 warnings about potentially uninitialized variables.
Added const version of Option::next().
```

**Comment by danoli3** [comment]
Okay going to test on Mac and VS manually before merging

**Comment by danoli3** [comment]
macOS all looking good!

VS Look good except addons include dirs. 
Patched that about to test again

**Comment by ofTheo** [comment]
whoa - that's a lot of changes! 
Thanks @danoli3 !!! does this fix #554 or #555 ?

**Comment by danoli3** [comment]
Yeah it does! Almost there, just fixing some last issues found with addon includes dirs

**Comment by ofTheo** [comment]
@dimitre want to take a look a these changes before merging?

**Comment by danoli3** [comment]
Having some issues with include directories for addons, it just adding every directory under the sun in those directories... adding a bit of logic to fix this

**Comment by danoli3** [comment]
Okay going to quickly cleanup the command stuff and get some gui side fixes in but tested addons in VS and macOS a lot and this is working finally

**Comment by danoli3** [comment]
@dimitre after this feel free to try and solve any wstring() potential issues on Windows, but for now, I'm not overthinking that

File: commandLine/src/addons/ofAddon.h [hunk 1]
```diff
@@ -176,15 +179,30 @@ class ofAddon {
...
 	string currentParseState { "" };
-
+	string emptyString = { "" };
 	void preParseConfig();
 	void parseConfig();
 	void parseVariableValue(const string & variable, const string & value, bool addToValue, const string & line, int lineNum);
-	void addReplaceString(string & variable, string value, bool addToVariable);
-	void addReplaceStringVector(vector<string> & variable, string value, string prefix, bool addToVariable);
-	void addReplaceStringVector(vector<LibraryBinary> & variable, string value, string prefix, bool addToVariable);
+	void parseVariableValuePath(std::filesystem::path & variable, const string & value, bool addToValue, const string & line, int lineNum);
```
Comment (dimitre) [inline commandLine/src/addons/ofAddon.h:186]: std::filesystem::path -> of::filesystem::path

**Comment by danoli3** [comment]
Solves so many issues with current PG.

**Comment by dimitre** [comment]
Great one! I've tested after merged, seems 100% here.

**Pull request merged**